Add continuous integration tests

BramBonne opened the pull request:
Runs tests on every push and pull request.

Additionally, this bumps the version number to trigger the tests to be picked up by AOSP's automatic importer.

Dushistov commented on .github/workflows/ci.yaml:
```diff
@@ -0,0 +1,18 @@
+name: Continuous integration
```
There is travis CI enabled for this repo: https://github.com/Nercury/android_logger-rs/blob/master/.travis.yml .
I suppose migration to github actions is ok, but we need at least convert all checks from travis to github actions.

Dushistov reviewed (changes requested):
There is travis CI enabled for this repo: https://github.com/Nercury/android_logger-rs/blob/master/.travis.yml .
I suppose migration to github actions is ok, but we need at least convert all checks from travis to github actions.

BramBonne commented:
Thanks for the review! I totally missed the existing Travis CI checks.

I moved everything over to GitHub actions for now, but please let me know if you prefer it when I'd just add the tests to the Travis CI instead.

Dushistov commented:
Looks like without enabled github actions for repo it is impossible to test your PR,
so I created #43 . It enabled github actions for this repo when merged, and you rebase your PR,
I suppose we see results of github actions for this PR.

BramBonne commented:
Sounds good!

Alternatlively, you can see my PR in action at https://github.com/BramBonne/android_logger-rs/actions as well, as it's been merged there already.

Dushistov reviewed (approved):
Not sure about version bump, this is depend on @Nercury . 
But otherwise looks good.

Nercury commented:
This should probably only need a patch version change if the API is exactly the same. But there aren't yet many users of `0.11` version, so it's fine in this case.

Nercury commented:
Let's also remove travis.

BramBonne commented:
Thanks for taking a look! I changed the new version number to 0.10.1 now.

As for Travis, I removed the `.travis.yml` file in the previous patch already; is any additional action required to prevent it from triggering?

Nercury commented:
I am going to sort that travis out.
